src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -244,31 +244,72 @@ static SymbolNamesWithValueOption<INamedTypeSymbol>.NameParts GetParts(string na
             }
         }
 
+        public static SymbolNamesWithValueOption<Unit> GetInheritanceExcludedSymbolNamesOption(
+            this AnalyzerOptions options,
+            DiagnosticDescriptor rule,
+            Compilation compilation,
+            string defaultForcedValue,
```

Comment:
There is actually a problem with having `defaultForcedValue` as a method parameter. Because of the caching mechanism if there are multiple calls for the same rule with different values only the first one will be taken into account. I am facing this problem when trying to mimic this for #3212